### DISCUSSION TITLE
How to lock InnerBlocks in registerBlockVariation()

### DISCUSSION THREAD

**Discussion by lindsayweb** [WordPress/gutenberg#60294] [Q&A]
I'm using the following media-text block variation, and I want to lock the InnerBlocks (not the block itself). Adding the templateLock attribute didn't work. Is it possible?

    wp.blocks.registerBlockVariation(
    'core/media-text',
    {
        name: 'media-text-offset',
        title: 'Offset Callout',
        attributes: {
            backgroundColor: 'orange',
            className: 'is-style-offset',
            templateLock: 'all'
        },
        isActive: [ 'className' ],
        innerBlocks: [
            ["core/paragraph", { placeholder: "Label", className: "label" }],
            ["core/heading", { level: 2, placeholder: "Heading" }],
            ["core/paragraph"]
        ],
    }
    );

**Comment by stokesman** [comment]
It’s not possible at the moment. Unlike the Group or Columns block `templateLock` is not an attribute of the Media & Text block. If someone were to make a PR to add the attribute I'd expect it would be merged. It should be quite simple. For reference, here’s the PR that did so for the aforementioned blocks #26128. It could probably be simpler than that because it’s only one block and I don’t suppose there’d be a need to add tests.
